client/my-sites/domains/domain-management/contacts-privacy/card.jsx

```diff
@@ -5,55 +5,136 @@
 	static propTypes = {
 		privateDomain: PropTypes.bool.isRequired,
+		privacyAvailable: PropTypes.bool.isRequired,
 		selectedDomainName: PropTypes.string.isRequired,
 		selectedSite: PropTypes.oneOfType( [ PropTypes.object, PropTypes.bool ] ).isRequired,
+		contactInfoDisclosureAvailable: PropTypes.bool.isRequired,
+		contactInfoDisclosed: PropTypes.bool.isRequired,
+		isPendingIcannVerification: PropTypes.bool.isRequired,
 	};
 
-	render() {
-		const { privateDomain, translate, selectedDomainName } = this.props;
-		let privacyText = translate(
-			'{{strong}}Privacy Protection is enabled{{/strong}} so your contact information' +
-				' {{strong}}is not shown{{/strong}}.',
-			{
-				components: {
-					strong: <strong />,
-				},
-			}
-		);
-		if ( ! privateDomain ) {
-			privacyText = translate(
-				'{{strong}}Privacy Protection is disabled{{/strong}} so your own contact information' +
-					' {{strong}}is shown{{/strong}}.',
+	togglePrivacy = () => {
+		const { selectedSite, privateDomain, selectedDomainName: name } = this.props;
+
+		if ( privateDomain ) {
+			this.props.disableDomainPrivacy( selectedSite.ID, name );
+		} else {
+			this.props.enableDomainPrivacy( selectedSite.ID, name );
+		}
+	};
+
+	toggleContactInfo = () => {
+		const { selectedSite, contactInfoDisclosed, selectedDomainName: name } = this.props;
+
+		if ( contactInfoDisclosed ) {
+			this.props.redactDomainContactInfo( selectedSite.ID, name );
+		} else {
+			this.props.discloseDomainContactInfo( selectedSite.ID, name );
+		}
+	};
+
+	getPrivacyProtection() {
+		const { privateDomain, privacyAvailable } = this.props;
+		const { translate, isUpdatingPrivacy } = this.props;
+
+		if ( ! privacyAvailable ) {
+			return false;
+		}
+
+		return (
+			<Property label={ translate( 'Privacy Protection' ) }>
 				{
-					components: {
-						strong: <strong />,
-					},
+					<FormToggle
+						wrapperClassName="edit__privacy-protection-toggle"
+						checked={ privateDomain }
+						toggling={ isUpdatingPrivacy }
+						disabled={ isUpdatingPrivacy }
+						onChange={ this.togglePrivacy }
+					/>
 				}
-			);
+			</Property>
+		);
+	}
+
+	getContactInfoDisclosed() {
+		const {
+			translate,
+			isUpdatingPrivacy,
+			privateDomain,
+			privacyAvailable,
+			contactInfoDisclosureAvailable,
+			contactInfoDisclosed,
+			isPendingIcannVerification,
+		} = this.props;
+
+		if ( ! privacyAvailable || ! contactInfoDisclosureAvailable || privateDomain ) {
+			return false;
 		}
 
+		const contactVerificationNotice = isPendingIcannVerification ? (
+			<div class="edit__disclose-contact-information-warning">
+				<Gridicon icon="info-outline" size={ 18 } />
+				<p>
+					{ translate(
+						'You need to verify the contact information for the domain before you can disclose it publicly.'
+					) }
+				</p>
+			</div>
+		) : null;
+
+		return (
+			<div>
+				<Property label={ translate( 'Display my contact information in public WHOIS' ) }>
+					<FormToggle
```

Comment:
Side note RE the `FormToggle` component -- one of the design folks pointed out that the label is normally on the right of the toggle. I researched more and found that the component README was inaccurate -- PR to fix that is in #37351. Not sure if we should fix that here too. (There is also the fact that `aria-label` is not specified below.)

I have no idea how (in)accessible the UI is already, so this is definitely non-blocking and more of a general finding.